point_viewer_proto_rust/src/proto.proto

```diff
@@ -75,23 +75,50 @@ message NodeId {
   int64 deprecated_index = 2;
 }
 
-message Node {
-  enum PositionEncoding {
+enum PositionEncoding {
     INVALID = 0;
     Uint8 = 1;
     Uint16 = 2;
     Float32 = 3;
     Float64 = 4;
-  }
+}
 
+
+message Node {
   PositionEncoding position_encoding = 2;
   int64 num_points = 3;
   NodeId id = 4;
 }
 
+enum AttributeData{
+    I64 = 0;
+    U64 = 1;
+    F32 = 2;
+    F64 = 3;
+    U8Vec3 = 4;
+    F64Vec3 = 5;
+}
+
+message Attribute{
+  string name = 1;
+  AttributeData attr_type = 2;
+}
+
+message S2Cell {
+  PositionEncoding position_encoding = 1;
+  uint64 cell_id = 2;
+  uint64 num_points = 3;
+}
+
+
 message Meta {
   int32 version = 1;
   AxisAlignedCuboid bounding_box = 4;
   double resolution = 3;
-  repeated Node nodes = 5;
+  repeated Node nodes = 5; 
+  repeated S2Cell cells = 6;
```

Comment:
Do you think a comment saying `nodes` and `cells` are mutually exclusive and used either for octrees or s2 cells would make sense?